res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,719 @@
+                        RED : OFF);
+                    if (samplerChan <= 4) { // Handle first 4 samplers in split mode
+                        // PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x14 - 1 + samplerChan, value ?
+                            RED : OFF);
+                        // SHIFT+PAD5 Mode A+B (sampler 1 in split mode)
+                        midi.sendShortMsg(i, 0x1C - 1 + samplerChan, value ?
+                            RED : OFF);
+                    }
+                }
+            }
+            break;
+    }
+};
+
+ReloopBeatmix24.SamplerPlay = function(value, group, control) {
+    var samplerRegEx = /\[Sampler(\d+)\]/;
+    var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+    if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
+        var ledColor;
+        if (value) {
+            ledColor = VIOLET;
+        } else {
+            ledColor = engine.getValue(group, "track_samples") ? RED : OFF;
+        }
+
+        // We need to switch off pad lights before changing color otherwise
+        // VIOLET to RED transition does not work well. (???)
+        for (var i = 0x91; i <= 0x94; i++) {
+            // PAD1 Mode A
+            midi.sendShortMsg(i, 0x08 - 1 + samplerChan, OFF);
+            midi.sendShortMsg(i, 0x08 - 1 + samplerChan, ledColor);
+            // SHIFT+PAD1 Mode A
+            midi.sendShortMsg(i, 0x48 - 1 + samplerChan, OFF);
+            midi.sendShortMsg(i, 0x48 - 1 + samplerChan, ledColor);
+            if (samplerChan <= 4) { // Handle first 4 samplers in split mode
+                // PAD5 Mode A+B (sampler 1 in split mode)
+                midi.sendShortMsg(i, 0x14 - 1 + samplerChan, OFF);
+                midi.sendShortMsg(i, 0x14 - 1 + samplerChan, ledColor);
+                // SHIFT+PAD5 Mode A+B (sampler 1 in split mode)
+                midi.sendShortMsg(i, 0x1C - 1 + samplerChan, OFF);
+                midi.sendShortMsg(i, 0x1C - 1 + samplerChan, ledColor);
+            }
+        }
+    }
+};
+
+ReloopBeatmix24.loopDefined = function(value, group, control) {
+    var channelRegEx = /\[Channel(\d+)\]/;
+    var channelChan = parseInt(channelRegEx.exec(group)[1]);
+    if (channelChan <= 4) {
+        midi.sendShortMsg(0x90 + channelChan, 0x44, value < 0 ? OFF :
+            VIOLET);
+    }
+};
+
+ReloopBeatmix24.startJogLedSpinnie = function(value, group, control) {
+
+    var chan = parseInt(group.substr(8, 1), 10);
+
+    // Stop and remove reference to old timer, shut off all leds,
+    // we reached the next beat
+    if (jogWheelTimers[group]) {
+        engine.stopTimer(jogWheelTimers[group]);
+        delete jogWheelTimers[group];
+    }
+    ReloopBeatmix24.AllJogLEDsToggle(0x90 + chan, OFF);
+
+    if (value) {
+        // Lit top led
+        midi.sendShortMsg(0x90 + chan, 0x30, ON);
+
+        // Set timer for next led in 2 seconds / 16 led intervals
```

Comment:
Thanks for the hint. I tried something like that with a variable timer based on calculation with duration and beat_active to always lit the top led when we are on a beat (thus having left  & right deck synchronized will lead to synchronized leds), but that didn't work very well and I reverted to a fixed revtime.
I keep that for after vacation ;)